core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -296,6 +302,81 @@ object UnitActions {
                 }.takeIf { unit.currentMovement > 0 && canPillage(unit, tile) })
     }
 
+    private fun pillageLooting(tile: TileInfo, unit: MapUnit) {
+        // Stats objects for reporting pillage results in a notification
+        val globalPillageYield = Stats()
+        val toCityPillageYield = Stats()
+        val closestCity = unit.civInfo.cities.minByOrNull { it.getCenterTile().aerialDistanceTo(tile) }
+        val improvement = tile.ruleset.tileImprovements[tile.improvement]!!
+
+        for (unique in improvement.getMatchingUniques(UniqueType.PillageYieldRandom)) {
+            for (stat in unique.stats) {
+                val looted = Random.nextInt((stat.value + 1).toInt()) + Random.nextInt((stat.value + 1).toInt())
+                // when statement handles different stat transactions
+                // gold, faith, culture, and science are given to the civ
+                // food and production are given to the nearest city
+                // happiness can't be pillaged
+                when (stat.key) {
+                    Stat.Gold, Stat.Faith, Stat.Science, Stat.Culture -> {
+                        unit.civInfo.addStat(stat.key, looted)
+                        globalPillageYield[stat.key] += looted.toFloat()
+                    }
+                    Stat.Food, Stat.Production -> {
+                        if (closestCity != null) {
+                            closestCity.addStat(stat.key, looted)
+                            toCityPillageYield[stat.key] += looted.toFloat()
+                        }
+                    }
+                    else -> {}
```

Comment:
Instead of having this logic twice, have it once.
You can do this in a couple of ways, I would probably take all random stats, randomize them, then add them to static stats and iterate final stat list.

Reply:
Ah I see what you mean.

Reply:
And wasn't there some code to support which stats have a global pool and can therefore be used for buying stuff? There may be something reusable there. And don't forget Stats is a nice and fast aggregator, so you could pull a Stats() add your loot to it _then_ forward the sums to civ/city pools _and_ use it to feed notifications - with consistent ordering regardless of unique order. (don't use values, that one includes zeroes. Normal iteration or asSequence won't)